janitor/utils.py

```diff
@@ -410,133 +410,104 @@ def _wrapped(x, *args, **kwargs):
     return _wrapped
 
 
-def _check_instance(entry: Dict):
+def _grid_computation(entry: Dict) -> pd.DataFrame:
     """
-    Function to check instances in the expand_grid function.
-    This checks if entry is a dictionary,
-    checks the instance of value in key:value pairs in entry,
-    and makes changes to other types as deemed necessary.
-    Additionally, ValueErrors are raised if empty containers are
-    passed in as values into the dictionary.
-    How each type is handled, and their associated exceptions,
-    are pretty clear from the code.
+    This function relies on itertools' `product` function
+    to create the different combinations. It is possible
+    that in the future, a faster implementation will be used,
+    and still respect data types.
+    In this function, a couple of checks are conducted as well.
+    This include type checking, as well as the size (if it is a
+    numpy array, it should not be greater than 2). Dataframes
+    with MultiIndex columns are not supported.
```

Comment:
As mentioned in the overall comment, marking out this paragraph to be changed to SEMBR-compliant form.